pkg/chunk/purger/blocks_purger_api.go

```diff
@@ -0,0 +1,116 @@
+package purger
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/oklog/ulid"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/thanos-io/thanos/pkg/objstore"
+
+	cortex_tsdb "github.com/cortexproject/cortex/pkg/storage/tsdb"
+	"github.com/cortexproject/cortex/pkg/tenant"
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+type BlocksPurgerAPI struct {
+	bucketClient objstore.Bucket
+}
+
+func NewBlocksPurgerAPI(storageCfg cortex_tsdb.BlocksStorageConfig, logger log.Logger, reg prometheus.Registerer) (*BlocksPurgerAPI, error) {
+	bucketClient, err := createBucketClient(storageCfg, logger, reg)
+	if err != nil {
+		return nil, err
+	}
+
+	return newBlocksPurgerAPI(bucketClient), nil
+}
+
+func newBlocksPurgerAPI(bkt objstore.Bucket) *BlocksPurgerAPI {
+	return &BlocksPurgerAPI{bucketClient: bkt}
+}
+
+func (api *BlocksPurgerAPI) DeleteTenant(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	userID, err := tenant.TenantID(ctx)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
```

Comment:
We return `http.StatusUnauthorized` is other APIs. We should be consistent here too.

_Same comment applies to `DeleteTenantStatus()`._

Reply:
I don't think this is correct. This endpoint uses authentication middleware, which already checks if there is correct x-scope-orgID header, and stores tenant into context. Auth middleware already returns `http.StatusUnauthorized`. If we get error here, it can be for different reasons, e.g. using multi-tenants.

Reply:
(Btw, I purger's other handler uses `http.StatusBadRequest`, instead of `http.StatusUnauthorized`. I copied it from there. I think it's more correct.)

Reply:
I see your point. I'm OK keeping it as is in this PR. However, I think it's worth opening a GitHub issue about it, cause we should keep a keep a consistent approach within Cortex.